x/vm/types/tx_args.go

```diff
@@ -282,3 +196,293 @@ func (args *TransactionArgs) GetData() []byte {
 	}
 	return nil
 }
+
+// CallDefaults sanitizes the transaction arguments, often filling in zero values,
+// for the purpose of eth_call class of RPC methods.
+func (args *TransactionArgs) CallDefaults(globalGasCap uint64, baseFee *big.Int, chainID *big.Int) error {
+	// Reject invalid combinations of pre- and post-1559 fee styles
+	if args.GasPrice != nil && (args.MaxFeePerGas != nil || args.MaxPriorityFeePerGas != nil) {
```

Comment:
seems we can remove https://github.com/cosmos/evm/blob/ca9ecfdaf039cfa620ab9f1a446c8c55aee38157/x/vm/types/tx_args.go#L89-L92 
since gasPrice with EIP-1559 fee param checking in args as params default setting should be performed in function `SetDefaults` before `ToMessage`